receiver/pulsarreceiver/config.go

```diff
@@ -0,0 +1,148 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pulsarreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/pulsarreceiver"
+
+import (
+	"errors"
+
+	"github.com/apache/pulsar-client-go/pulsar"
+	"go.opentelemetry.io/collector/config"
+)
+
+type Config struct {
+	config.ReceiverSettings `mapstructure:",squash"`
+	// Configure the service URL for the Pulsar service.
+	Endpoint string `mapstructure:"endpoint"`
+	// The topic of pulsar to consume logs,metrics,traces. (default = "otlp_traces" for traces,
+	// "otlp_metrics" for metrics, "otlp_logs" for logs)
+	Topic string `mapstructure:"topic"`
+	// The Subscription that receiver will be consuming messages from (default "otlp_subscription")
+	Subscription string `mapstructure:"subscription"`
+	// Encoding of the messages (default "otlp_proto")
+	Encoding string `mapstructure:"encoding"`
+	// Name specifies the consumer name.
+	ConsumerName string `mapstructure:"consumer_name"`
+	// Set the path to the trusted TLS certificate file
+	TLSTrustCertsFilePath string `mapstructure:"tls_trust_certs_file_path"`
+	// Configure whether the Pulsar client accept untrusted TLS certificate from broker (default: false)
+	Insecure       bool           `mapstructure:"insecure"`
```

Comment:
I believe it's `TLSAllowInsecureConnection` in the exporter. Let's keep it consistent